src/Model.jl

```diff
@@ -65,32 +65,64 @@ end
+Arguments
+- `lattice::AbstractMatrix`: Lattice vectors given column-wise (atomic units).
+- `atoms::Vector{<:Element}`: Atomic species (may include pseudopotential metadata).
+- `positions::Vector{Vec3}`: Atomic positions in fractional (reduced) coordinates.
 
-`magnetic_moments` is only used to determine the symmetry and the
-`spin_polarization`; it is not stored inside the datastructure.
-
-`smearing` is Fermi-Dirac if `temperature` is non-zero, none otherwise
-
-The `symmetries` kwarg allows (a) to pass `true` / `false` to enable / disable
-the automatic determination of lattice symmetries or (b) to pass an explicit list
-of symmetry operations to use for lowering the computational effort.
-The default behaviour is equal to `true`, namely that the code checks the
-specified model in form of the Hamiltonian `terms`, `lattice`, `atoms` and `magnetic_moments`
-parameters and from these automatically determines a set of symmetries it can safely use.
-If you want to pass custom symmetry operations (e.g. a reduced or extended set) use the
-`symmetry_operations` function. Notice that this may lead to wrong results if e.g. the
-external potential breaks some of the passed symmetries. Use `false` to turn off
-symmetries completely.
+## Keyword arguments
+- `model_name::String` (default: "custom"): Human-readable model name (e.g. "LDA", "PBE").
+- `εF` (default: `nothing`): Fixed Fermi level option. If set, the model is defined at fixed
+    Fermi level instead of fixed electron count. Mutually exclusive with `n_electrons`.
+- `n_electrons::Union{Int,Nothing}` (default: inferred from `atoms`): Total electron count.
+    If `εF !== nothing` then `n_electrons` must be `nothing`.
+- `disable_electrostatics_check::Bool` (default: `all(iszero, charge_ionic.(atoms))`): When
+    `false` (the default safety path), constructing a non-neutral model will raise an error
+    unless explicitly allowed. Set to `true` to bypass neutrality/electrostatics
+    checks (experimental).
+- `magnetic_moments` (default: `[]`): Per-atom initial magnetic moments used to infer
+    `spin_polarization` and to determine symmetries. These values influence model setup
+    but are otherwise not stored on the `Model`.
+- `terms::Vector` (default: `[Kinetic()]`): List of energy term types to include in the
+    Hamiltonian.
+- `temperature::Real` (default: `0`): Electronic temperature.
+    If `> 0`, fractional occupations are enabled and `smearing` defaults to
+    [`Smearing.FermiDirac`](@ref)().
+- `smearing::Smearing.SmearingFunction` (default: `FermiDirac()` when `temperature>0`,
+    else `None()`):
+    Function used to convert eigenvalue differences to occupations.
+- `spin_polarization::Symbol` (default: `determine_spin_polarization(magnetic_moments)`):
+    Controls spin treatment; allowed values are `:none`, `:collinear`, `:spinless`.
+    (`:full` exists but is not supported and will error.)
+- `symmetries::Union{Bool,Vector{SymOp}}` (default: `true`):
```

Comment:
Defaults and type is sort of true, but not really. I understand why you did that, but I do wonder if it would not be better to be less specific and refer to the function (mentioning that usually the result is `true`) and then indeed explaining the three cases as you do, but without annotating a type.

Reply:
Ok, I removed the type annotation and link to the `default_symmetries` function